### PR TITLE
Add uuid functions

### LINKED ISSUE #3372: Add UUID relevant functions
**Summary**

- Generate UUID v4
- Check if UUID is empty
- Check if UUID is not empty

### PR DESCRIPTION
**Pull request by lianghanzhen**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Add uuid functions:
- generateUUIDv4
- zeroUUID
- isemptyUUID
- isnotemptyUUID

## Changelog

- New Feature

## Related Issues

Fixes #3372

## Test Plan

Unit Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
It would be nice to add some uuid stateless test :)

File: common/functions/src/scalars/uuids/uuid_verifier.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                    }
+                }
+
+                Ok(DataColumn::Constant(
+                    DataValue::Boolean(Some(true)),
+                    input_rows,
+                ))
+            }
+            _ => Ok(DataColumn::Constant(
+                DataValue::Boolean(Some(true)),
```
Comment (sundy-li) [inline common/functions/src/scalars/uuids/uuid_verifier.rs:138]: Better to have `T::default_verify`

Comment (lianghanzhen) [reply common/functions/src/scalars/uuids/uuid_verifier.rs]: Added.

**Comment by lianghanzhen** [comment]
Added stateless tests for uuid functions.

**NOTE**: the test for `generateUUIDv4` function is missing because the value is not fixed which is impossible to verify.

**Comment by sundy-li** [comment]
> NOTE: the test for generateUUIDv4 function is missing because the value is not fixed which is impossible to verify.

You can:
```
select length(  generateUUIDv4() ) = xxx;
select ignore( generateUUIDv4() );
```

**Comment by lianghanzhen** [comment]
Still can not verify the content, but better than nothing.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**